tests/sentry/incidents/test_models.py

```diff
@@ -350,6 +351,24 @@ def test(self):
         assert incident.duration == timedelta(minutes=2)
 
 
+class IncidentAlertRuleRelationTest(TestCase):
+    def test(self):
+        self.alert_rule = self.create_alert_rule()
+        self.trigger = self.create_alert_rule_trigger(self.alert_rule)
+        self.incident = self.create_incident(alert_rule=self.alert_rule, projects=[self.project])
+
+        assert self.incident.alert_rule.id == self.alert_rule.id
+        all_alert_rules = list(AlertRule.objects.all())
+        assert self.alert_rule in all_alert_rules
+
+        self.alert_rule.status = AlertRuleStatus.ARCHIVED.value
+        self.alert_rule.save()
+
+        all_alert_rules = list(AlertRule.objects.all())
+        assert self.alert_rule not in all_alert_rules
+        assert self.incident.alert_rule.id == self.alert_rule.id
```

Comment:
Test looks good! I don't think we need to necessarily merge this since it's just testing Django functionality, I just wanted to confirm that it's working as we expect.